GetServices not working on Centos 7

The SystemInfoTest does not seem to work on my Centos 7 install (on latest master commit `d7d2e9f8dd46dd4ce6dd1e787a15f4ce29626d10`).

```
$ ./mvnw test-compile -pl oshi-core -q exec:java   -Dexec.mainClass="oshi.SystemInfoTest"   -Dexec.classpathScope="test"
2019-12-03 13:21:40.938 [ERROR] LinuxOperatingSystem - Directory: /etc/init does not exist
2019-12-03 13:21:41.067 [ERROR] LinuxOperatingSystem - Directory: /etc/init does not exist
```

```
$ uname -a
Linux my-host 3.10.0-1062.4.3.el7.x86_64 #1 SMP Wed Nov 13 23:58:53 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
```

Version 3.13.4 seems to work though:
```
$ ./mvnw test-compile -pl oshi-core -q exec:java   -Dexec.mainClass="oshi.SystemInfoTest"   -Dexec.classpathScope="test"
2019-12-03 13:28:48.488 [INFO] SystemInfoTest - Initializing System...
GNU/Linux CentOS Linux 7.7.1908 (Core) build 3.10.0-1062.4.3.el7.x86_64
2019-12-03 13:28:48.526 [INFO] SystemInfoTest - Checking computer system...
...
```

--- a/oshi-core/src/main/java/oshi/software/os/linux/LinuxOperatingSystem.java
+++ b/oshi-core/src/main/java/oshi/software/os/linux/LinuxOperatingSystem.java
@@ -749,21 +749,39 @@ public class LinuxOperatingSystem extends AbstractOperatingSystem {
             services.add(s);
             running.add(p.getName());
         }
-        // Get Directories for stopped services
-        File dir = new File("/etc/init");
-        if (dir.exists() && dir.isDirectory()) {
-            for (File f : dir.listFiles((f, name) -> name.toLowerCase().endsWith(".conf"))) {
-                // remove .conf extension
-                String name = f.getName().substring(0, f.getName().length() - 5);
+        boolean systemctlFound = false;
+        List<String> systemctl = ExecutingCommand.runNative("systemctl list-unit-files");
+        for (String str : systemctl) {
+            String[] split = ParseUtil.whitespaces.split(str);
+            if (split.length == 2 && split[0].endsWith(".service") && "enabled".equals(split[1])) {
+                // remove .service extension
+                String name = split[0].substring(0, split[0].length() - 8);
                 int index = name.lastIndexOf('.');
                 String shortName = (index < 0 || index > name.length() - 2) ? name : name.substring(index + 1);
                 if (!running.contains(name) && !running.contains(shortName)) {
                     OSService s = new OSService(name, 0, STOPPED);
                     services.add(s);
+                    systemctlFound = true;
                 }
             }
-        } else {
-            LOG.error("Directory: /etc/init does not exist");
+        }
+        if (!systemctlFound) {
+            // Get Directories for stopped services
+            File dir = new File("/etc/init");
+            if (dir.exists() && dir.isDirectory()) {
+                for (File f : dir.listFiles((f, name) -> name.toLowerCase().endsWith(".conf"))) {
+                    // remove .conf extension
+                    String name = f.getName().substring(0, f.getName().length() - 5);
+                    int index = name.lastIndexOf('.');
+                    String shortName = (index < 0 || index > name.length() - 2) ? name : name.substring(index + 1);
+                    if (!running.contains(name) && !running.contains(shortName)) {
+                        OSService s = new OSService(name, 0, STOPPED);
+                        services.add(s);
+                    }
+                }
+            } else {
+                LOG.error("Directory: /etc/init does not exist");
+            }
         }
         return services.toArray(new OSService[0]);
     }

